corehq/motech/openmrs/atom_feed.py

```diff
@@ -372,3 +362,31 @@ def fields_from_observations(observations, mappings):
             xmlns=XMLNS_OPENMRS,
             device_id=OPENMRS_ATOM_FEED_DEVICE_ID + repeater.get_id,
         )
+
+
+def get_updates_from_observations(observations, mappings):
+    """
+    Traverse a tree of observations, and return the ones mapped to case
+    properties.
+    """
+    fields = {}
+    for obs in observations:
+        if obs['concept']['uuid'] in mappings:
```

Comment:
nit: Perhaps this dict would always have these keys, but could be made future-proof by `obs.get('concept', {}).get('uuid')`

Reply:
A counter-argument is that if those keys do not exist then there may be a bug in the system and a hard failure may be better than passing silently without error? I don't have enough context to know which is better.

Reply:
I hear you. Your point adds support for checking the schema of both outgoing requests, and incoming responses.

The best candidate I've found for that so far is the [schema](https://pypi.org/project/schema/) library, but I'm open to other suggestions.